DEPENDENCIES.md

```diff
@@ -0,0 +1,58 @@
+# Dependency Management
+[`dep`](https://github.com/golang/dep) is a tool of choice when it comes to dependency management.
+
+## How we use `dep`.
+
+1. Transitive dependencies of `go-ethereum`. The most important thing for us is
+   to be in-sync there. We want to reduce the regression scope. 
+   Hence, we pin down all the dependencies of `go-ethereum` with SHAs in `Gopkg.toml` when
+   importing a new version of upstream. (This is considered a bad practice for
+   `dep` but we are willing
+
+2. Exclusive `status-go` dependencies. The policy there is to keep them as
+   fresh as possible. Hence, no constraints for them in the `toml` file.
+
+## Installing `dep`
+
+`go get -u github.com/golang/dep/cmd/dep`
+
+
+## Docs (worth reading)
+1. [README](https://github.com/golang/dep/blob/master/README.md)
+2. [F.A.Q.](https://github.com/golang/dep/blob/master/docs/FAQ.md)
+
+
+## Adding a new Dependency
+(see [Adding a new dependency](https://github.com/golang/dep#adding-a-dependency))
+1. `$ dep ensure -add github.com/foo/bar`
+2. Commit changes.
+
+
+## Updating a dependency
```

Comment:
i think it worth noting how to install dependencies from scratch, e.g dep ensure

Reply:
Do we ever get a repo state when you need to install all deps from scratch? (aka with empty `vendor`)?

Reply:
i guess you are right

Reply:
fixed in the latest push